compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1585,11 +1602,20 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
         if args.len() < sig.inputs().len() || (args.len() > sig.inputs().len() && !sig.c_variadic) {
             span_mirbug!(self, term, "call to {:?} with wrong # of args", sig);
         }
+
+        let func_ty = if let TerminatorKind::Call { func, .. } = &term.kind {
+            Some(func.ty(body, self.infcx.tcx))
+        } else {
+            None
+        };
```

Comment:
This is slightly shorter:

```suggestion
        let func_ty = match &term.kind {
            TerminatorKind::Call { func, .. } => Some(func.ty(body, self.infcx.tcx)),
            _ => None,
        };
```